### ISSUE TITLE
Support sensitive options (like password) which are read from standard input

### ISSUE THREAD

**Issue by lenkite** [remkop/picocli#374]
This is useful since folks nowadays have persistent history and CLI tool security best practice strongly recommends not providing passwords at the command line. Passwords if needed should be read from standard input and masked.

```java
class ListNodesCmd {

    @Option(names = "-u", description = "Controller user")
    String  user;

    @Option(names = "-p", description = "Controller password", sensitive=true, prompt="Enter Controller Password")
    char[]  password;

    @Option(names="-c", description="controller host", required=true)
    URI controllerHost
     ...
}
```

```
> list-nodes -c https://management-node  -u john -p 
Enter Controller Password: *****
...
```

**Comment by remkop** [comment]
I agree. I created ticket #82 for this some time ago but I’ve prioritized other tickets until now. Pull requests are always welcome!

**Comment by bobtiernay-okta** [comment]
This would be super useful! Also, right now there is no guidance in _not_ doing the natural thing of creating a `--password` argument, which is insecure given that it shows up in the process listing.

**Comment by remkop** [comment]
I finally addressed #82 and this will be in the upcoming v3.5 release.
